### ISSUE TITLE
A question about dataloader.py

### ISSUE THREAD

**Issue by si520** [Ruifeng-Tan/BatteryLife#6]
Thank you for the author's open-source work. I have a question : why build attention masks? I think the model can use all the data from the early cycles.

**Comment by Ruifeng-Tan** [comment]
Thank you for the question. This work is interested in developing a model that can predict the cycle life using the first $$N$$ ($$N\leq 100, N\in Z^+$$) cycles instead of using a fixed cycle number. In this vein, each battery provides 100 samples. The attention masks are used to generate samples with different usable numbers of cycles (e.g., 1, 2, 3, ..., 100) from each battery. We will use the attention masks to mask the data from the cycles that are not usable using code like `cycle_curve_data[tmp_curve_attn_mask==0] = 0 # set the unseen data as zeros` at the beginning of the model forward function. 

For example, when only the first 2 cycles are usable for a sample, the attention mask will be [1, 1, 0, 0, ..., 0, 0] for this sample.

We also notice that implementing masking in the `collate_fn` would be more beneficial for understanding, rather than in the `model forward function`. We are considering making changes to the code to facilitate code readability.

If you have further questions regarding the attention masks, please post them here.

**Comment by si520** [comment]
Thank you for your reply. I understand the process of building the dataset. In theory, using more cycles may lead to more accurate results. Would it be more appropriate to consider conducting the experiments separately? For example, comparing the performance of models on different datasets using only one cycle or ten cycles. This is just my personal suggestion for your reference. Thank you for your work.
